### PR TITLE
Added Umbraco Forms Heartcore API documentation

### PR DESCRIPTION
**Pull request by rasmusjp**
- [x] API Documentation
- [x] Usage Documentation
- [x] REST API examples
- [x] .NET Core examples
- [x] NodeJS examples

### CODE REVIEW

File: Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
+# Content Management API for Umbraco Forms
+
+**BASE URL**: `https://api.umbraco.io`
+
+**API version** 2.1
+
+:::note
+Umbraco Forms is not available in all plans, see the [Pricing & Features](https://umbraco.com/umbraco-heartcore-pricing/) for an overview of which plans it's included in.
```
Comment (sitereactor) [inline Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md]: Suggested change:
The availability of Umbraco Forms depend on the plan, see the [Pricing & Features](https://umbraco.com/umbraco-heartcore-pricing/) for an overview of which plans includes Forms.

File: Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+  "_links": {
+    "self": {
+      "href": "/api/forms/2edaf583-cf66-4d57-930c-f0772c3d1c52"
+    }
+  }
+}
+```
+
+## Submit entry
+
```
Comment (sitereactor) [inline Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md]: Missing some intro/summary type text like:

Submit form entries for a specific Form - this is what you would use when a form is submitted from your presentation layer.

**Comment by sitereactor** [comment]
FYI just rebased this branch off master

File: Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md (hunk 1 above)
Comment (rasmusjp) [reply Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md]: Changed

File: Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md (hunk 2 above)
Comment (rasmusjp) [reply Umbraco-Heartcore/API-Documentation/Content-Management/forms/index.md]: I've added the suggested text

File: Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+In the Forms section of the Umbraco backoffice is where you will find the Forms Builder, which allows you to create forms by adding Fields, Placeholder texts, Validation and Conditions.
+The purpose of this article is not to describe the Forms functionality itself, but to show how to use the APIs which are available for Umbraco Heartcore projects. If you want to learn the basics of 'Creating a form' before going further we recommend that you start with the [Umbraco Forms documentation section](https://our.umbraco.com/documentation/Add-ons/UmbracoForms/Editor/Creating-a-Form/).
+
+Before you continue with the rest of this article we recommend that you have at least one Form available that can be used for examples below. Here we will use a Forms with a Name field, an Email field and of course the Data Consent field, which is standard for all new Forms.
+
+## Retrieving a Form
+
+### Retrieving a Form using the REST API
+
+The API for Forms lives under the Content Management API, so a Bearer token and an API-Key is required to call all the Forms related endpoints.
```
Comment (rasmusjp) [inline Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md]:
```suggestion
The API for Forms lives under the Content Management API, so a Bearer token or an API-Key is required to call all the Forms related endpoints.
```

File: Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+```
+> Install-Package Umbraco.Headless.Client.Net
+```
+
+When using the library you need the Content Management part in order to work with Forms. New up the `ContentManagementService` and pass in the name of your Umbraco Heartcore project and either Username + Password of a backoffice user or an API-Key. In the example below we use an API-Key when retriving all available form definitions:
+
+```csharp
+var managementService = new ContentManagementService("project-alias", "api-key-value");
+var forms = await managementService.FormService.GetAll();
```
Comment (rasmusjp) [inline Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md]:
```suggestion
var forms = await managementService.Forms.GetAll();
```

I've changed the property name to `Forms` in the client

File: Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+```
+
+If you want to retrieve a specific form you can use the GetById method along with a guid id as shown below:
+
+```csharp
+var managementService = new ContentManagementService("project-alias", "api-key-value");
+var contactForm = await managementService.FormService.GetById(new Guid("0134604b-f583-4ebc-a3b6-c26ce0f1a11b"));
```
Comment (rasmusjp) [inline Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md]:
```suggestion
var contactForm = await managementService.Forms.GetById(new Guid("0134604b-f583-4ebc-a3b6-c26ce0f1a11b"));
```

Same as above

File: Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+```csharp
+var entry = new Dictionary<string, object>
+             {
+                 {"name", "John Smith"},
+                 {"email", "johnsmith@example.org"},
+                 {"dataConsent", true}
+             };
+
+var managementService = new ContentManagementService("project-alias", "api-key-value");
+await managementService.FormService.SubmitEntry(new Guid("0134604b-f583-4ebc-a3b6-c26ce0f1a11b"), entry);
```
Comment (rasmusjp) [inline Umbraco-Heartcore/Getting-Started-Cloud/Using-The-Forms-API/index.md]:
```suggestion
await managementService.Forms.SubmitEntry(new Guid("0134604b-f583-4ebc-a3b6-c26ce0f1a11b"), entry);
```

Same as above

**Comment by sitereactor** [comment]
Adding my :shipit: stamp of approval

@sofietoft not sure if you want to read through this and let us know if there is anything you want changed? This is scheduled for release Tuesday 17/12.

**Pull request merged**